data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpSinkResponseHandler.java

```diff
@@ -70,8 +71,9 @@ public HttpSinkResponseHandler(
     if (event == null) {
       return Future.failedFuture(new IllegalArgumentException("event cannot be null"));
     }
-    // TODO set the correct producer record key
-    return Future.succeededFuture(KafkaProducerRecord.create(topic, "", event));
+
+    final var recordKey = PartitionKey.extract(event);
+    return Future.succeededFuture(KafkaProducerRecord.create(topic, recordKey, event));
```

Comment:
the `KafkaProducerRecord` can handle null keys. In fact with null key it implements a partitioning strategy, while with `""` key it ends up putting everything in the same partition